### PR TITLE
Ansible modules for PowerScale release version 2.1.0

### PR DESCRIPTION
**Pull request by Bhavneet-Sharma**
# Description

- Added support for deleting an access zone and reordering the authentication providers in access zone module.

- Added support for service principal names(SPN) in AD module.

- Added support for SmartConnect zone alaises(DNS names) in network pool module.

- Create, modify, get details and delete an S3 bucket.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/ansible-powerscale/issues/55|

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, pep8, linting, or security issues
- [X] I have performed Ansible Sanity test using --docker default
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [X] Unit Testing
- [X] Functional Testing
- [X] Regression Testing

### CODE REVIEW

**Review by Pavan-Mudunuri — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -24,22 +24,22 @@ The Ansible Modules for Dell PowerScale support the following features:
 - Create, delete Filepool policy and Storagepool tiers.
 - Create, modify, get and delete NFS aliases.
 - Get attributes and entities of the array.
-- Find and close the SMB open files.
```
Comment (trisha-dell) [inline README.md:27]: revert this.

File: docs/Release Notes.md [hunk 2]
```diff
@@ -123,11 +124,14 @@ This section lists the limitations in this release of Ansible Modules for Dell P
 - No support for advanced PowerScale features
   - Advanced PowerScale features include tiering, replication, and so on.
 
+- AD
```
Comment (trisha-dell) [inline docs/Release Notes.md:127]: revert this.

File: playbooks/modules/smb.yml [hunk 3]
```diff
@@ -8,7 +8,7 @@
     verify_ssl: false
     api_user: 'user'
     api_password: 'password'
-    share_name: 'test_sample_smb'
+    smb_name: 'test_sample_smb'
```
Comment (trisha-dell) [inline playbooks/modules/smb.yml:11]: revert these.

Comment (Bhavneet-Sharma) [reply playbooks/modules/smb.yml:11]: Due to the same variable name as the parameter name, was getting a sanity error for this.
Errors were the following:

`var-naming[no-reserved]: Variables names must not be Ansible reserved names. (name)
playbooks/modules/smb.yml:11

var-naming[no-reserved]: Variables names must not be Ansible reserved names. (name)
playbooks/modules/snapshotschedule.yml:10

var-naming[no-reserved]: Variables names must not be Ansible reserved names. (name)
playbooks/modules/synciqreports.yml:13

var-naming[no-reserved]: Variables names must not be Ansible reserved names. (name)
playbooks/modules/synciqtargetreports.yml:13

var-naming[read-only]: This special variable is read-only. (role_name)
playbooks/modules/user.yml:[24](https://github.com/dell/ansible-powerscale/actions/runs/5656537467/job/15323781425?pr=59#step:8:25)`

Hence, I renamed this variable.

File: playbooks/modules/synciqreports.yml [hunk 4]
```diff
@@ -10,7 +10,7 @@
     api_password: 'password'
     state_present: 'present'
     id: '1-Test_syncIQ_policy'
-    report_name: 'Test_snap_schedule_123'
+    schedule_name: 'Test_snap_schedule_123'
```
[the remark this answers is not in the document]
Comment (Bhavneet-Sharma) [reply playbooks/modules/synciqreports.yml:13]: Due to the same variable name as the parameter name, was getting a sanity error for this.
Errors were the following:

```
var-naming[no-reserved]: Variables names must not be Ansible reserved names. (name)
playbooks/modules/smb.yml:11
var-naming[no-reserved]: Variables names must not be Ansible reserved names. (name)
playbooks/modules/snapshotschedule.yml:10

var-naming[no-reserved]: Variables names must not be Ansible reserved names. (name)
playbooks/modules/synciqreports.yml:13

var-naming[no-reserved]: Variables names must not be Ansible reserved names. (name)
playbooks/modules/synciqtargetreports.yml:13

var-naming[read-only]: This special variable is read-only. (role_name)
playbooks/modules/user.yml:24 
```

Hence, I renamed this variable.

File: playbooks/modules/synciqtargetreports.yml [hunk 5]
```diff
@@ -10,7 +10,7 @@
     api_password: 'password'
     state_present: 'present'
     id: '2-sample_policy'
-    target_report_name: 'sample_policy'
+    synciq_policy_name: 'sample_policy'
```
[the remark this answers is not in the document]
Comment (Bhavneet-Sharma) [reply playbooks/modules/synciqtargetreports.yml:13]: Due to the same variable name as the parameter name, was getting a sanity error for this.
Errors were the following:

```
var-naming[no-reserved]: Variables names must not be Ansible reserved names. (name)
playbooks/modules/smb.yml:11
var-naming[no-reserved]: Variables names must not be Ansible reserved names. (name)
playbooks/modules/snapshotschedule.yml:10

var-naming[no-reserved]: Variables names must not be Ansible reserved names. (name)
playbooks/modules/synciqreports.yml:13

var-naming[no-reserved]: Variables names must not be Ansible reserved names. (name)
playbooks/modules/synciqtargetreports.yml:13

var-naming[read-only]: This special variable is read-only. (role_name)
playbooks/modules/user.yml:24 
```

Hence, I renamed this variable.

File: docs/Release Notes.md [hunk 6]
```diff
@@ -100,8 +101,8 @@ This section lists the limitations in this release of Ansible Modules for Dell P
  
 - Users and Groups 
   - Only local users and groups can be created. 
-  - Operations on users and groups with very long names may fail.
-  - Modification of user password fails for OneFS version 9.5.
+  -  Operations on users and groups with very long names may fail.
+  - Modification of user may fail in some scenarios for OneFS version 9.5.
```
[the remark this answers is not in the document]
Comment (Bhavneet-Sharma) [reply docs/Release Notes.md]: Fixed,

File: README.md (hunk 1 above)
Comment (Bhavneet-Sharma) [reply README.md:27]: Fixed,

File: docs/Release Notes.md (hunk 2 above)
Comment (Bhavneet-Sharma) [reply docs/Release Notes.md]: Fixed,

File: CHANGELOG.rst [hunk 7]
```diff
@@ -4,22 +4,37 @@ Dellemc.Powerscale Change Logs
...
-- Added support for "everyone" user in filesystem module.
+- Added support for everyone user in filesystem module.
 - Added support for listing LDAP auth providers and user mapping rules in Info module.
 - Added support for PowerScale OneFS 9.5 Islander release.
 - For the execution of the PowerScale Ansible modules, python library "isilon-sdk" needs to be installed.
 
 New Modules
 -----------
 
-- dellemc.powerscale.user_mapping_rule - Create, modify, get details, and delete a user mapping rule.
```
[the remark this answers is not in the document]
Comment (Bhavneet-Sharma) [reply CHANGELOG.rst]: Fixed.

**Review by trisha-dell — DISMISSED** [review]

**Review by trisha-dell — DISMISSED** [review]

**Review by Pavan-Mudunuri — DISMISSED** [review]

**Review by Jennifer-John — DISMISSED** [review]

**Review by kuttattz — DISMISSED** [review]

**Review by Jennifer-John — DISMISSED** [review]

**Review by Pavan-Mudunuri — DISMISSED** [review]

**Review by kuttattz — APPROVED** [review]

**Review by Pavan-Mudunuri — APPROVED** [review]

**Review by Jennifer-John — APPROVED** [review]

**Review by trisha-dell — APPROVED** [review]

**Pull request merged**